scripts/cert_generation.sh

```diff
@@ -8,19 +8,40 @@ CA_PATH=${CA_PATH:-$WORKING_DIR/ca.crt}
 LOG_STORE=$3
 REGENERATE_NEEDED=0
 
+function info() {
+  local msg=$1
+  echo "[INFO] ${msg}" >> ${WORKING_DIR}/action.log
+}
+
 function init_cert_files() {
 
   if [ ! -f ${WORKING_DIR}/ca.db ]; then
+    info "Initializing missing file: ${WORKING_DIR}/ca.db"
     touch ${WORKING_DIR}/ca.db
   fi
 
   if [ ! -f ${WORKING_DIR}/ca.serial.txt ]; then
+    info "Initializing missing file: ${WORKING_DIR}/ca.serial.txt"
     echo 00 > ${WORKING_DIR}/ca.serial.txt
   fi
 }
 
 function generate_signing_ca() {
-  if [ ! -f ${WORKING_DIR}/ca.crt ] || [ ! -f ${WORKING_DIR}/ca.key ] || ! openssl x509 -checkend 0 -noout -in ${WORKING_DIR}/ca.crt; then
+  local regen=false
+  if [ ! -f ${WORKING_DIR}/ca.crt ] || [ "$(file -b ${WORKING_DIR}/ca.crt)" != "PEM certificate" ] ; then
```

Comment:
You don't have to do the first check since the second one will fail if the file is missing. You can change this to:

```
if [ "$(file -b ${WORKING_DIR}/ca.crt)" != "PEM certificate" ] ; then
```